test/trap.requestAnimationFrame.test.js

```diff
@@ -0,0 +1,139 @@
+import '@testing-library/jest-dom';
+import fetch, { disableFetchMocks, enableFetchMocks } from 'jest-fetch-mock';
+
+import trap from '../src/trap';
+import { REQUEST_ANIMATION_FRAME_MESSAGE_TYPE } from '../src/constants';
+
+const initialHtml = '<html><head></head><body>some text</body></html>';
+
+describe('browser with requestAnimationFrame callbacks', () => {
+  beforeAll(() => {
+    document.body.innerHTML = initialHtml;
+
+    // Here you must use `jest.resetModules` because otherwise Jest will have
+    // cached `trap.js` and it will _not_ run again.
+    jest.resetModules();
+
+    // Use fake timers -- it means that timers must be advanced manually!
+    jest.useFakeTimers();
+
+    // Enable fetch() mocks
+    enableFetchMocks();
+
+    // Set up fetch() mocks
+    fetch.mockResponse(() => Promise.resolve({ result: 'ok' }));
+  });
+
+  beforeEach(() => {
+    let count = 0;
+    // Mock requestAnimationFrame
+    jest.spyOn(window, 'requestAnimationFrame')
+      .mockImplementation((callback) => {
+        count += 1;
+        setTimeout(() => callback(count * 100), 100);
```

Comment:
I'd use something similar to the browsers' 60Hz timers, **OR** make it clear that this mock implements a 10Hz timeout, so later -- in the test -- the test will expect 2 messages if there is a 250ms timeout applied.

Reply:
Done